flake.nix

```diff
@@ -13,57 +12,63 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
         inherit system;
         overlays = [(import rust-overlay)];
-      };
+      });
+    gitRev = self.rev or self.dirtyRev or null;
+  in {
+    packages = eachSystem (system: {
+      helix = pkgsFor.${system}.callPackage ./default.nix {inherit gitRev;};
+      /**
+      The default Helix build. Uses the latest stable Rust toolchain, and unstable
+      nixpkgs.
+
+      The build inputs can be overriden with the following:
+
+      packages.${system}.default.override { rustPlatform = newPlatform; };
+
+      Overriding a derivation attribute can be done as well:
 
+      packages.${system}.default.overrideAttrs { buildType = "debug"; };
+      */
+      default = self.packages.${system}.helix;
+    });
+    checks = eachSystem (system: let
```

Comment:
For outputs that commonly access both `system` and `pkgs`, just use `lib.mapAttrs (system: pkgs: {...}) pkgsFor`.

Reply:
It's not simpler when function bodies grow and mention what needs to be named `pkgs` multiple times. Then `pkgsFor.${system}` becomes redundant, and then you would be foolish to not choose between `mapAttrs` and `let pkgs = pkgsFor.${system};`.

Speed does not matter, this decision pales in comparison to the evaluation time as the rest of the business logic. It is about readability. Choose one of the above, there is no guarantee that there will always be one check.